test/version_commands.bats

```diff
@@ -21,6 +21,9 @@ setup() {
   mkdir -p $CHILD_DIR
 
   cd $PROJECT_DIR
+
+  # asdf lib needed to run asdf.sh
+  cp -rf $BATS_TEST_DIRNAME/../{bin,lib} $ASDF_DIR/
```

Comment:
Do the tests fail without this change? Which new codepath requires this as I don't see any test changes.

Reply:
Yes, tests fail without this change because `setup_asdf_dir` creates an "empty" ASDF_DIR. It's more like a valid ASDF_DATA_DIR than a valid ASDF_DIR, because the {bin,lib} files are not there. Since with this change the asdf.sh will respect any previously set ASDF_DIR variable, that ASDF_DIR will be used, and the tests fail.

Maybe we should move this `cp -rf $BATS_TEST_DIRNAME/../{bin,lib} $ASDF_DIR/` (which is repeated may times after calling `setup_asdf_dir` in individual tests, so this function guarantees that a complete ASDF_DIR was created. WDYT?

Reply:
I think I understand after looking through some code. `setup_asdf_dir` makes sense, but I question why it would be needed even after your change. As I understand, it will not overwrite `ASDF_DIR` if it is already set. `ASDF_DIR` is set during the `setup_asdf_dir` func, so why would these `version_commands.bats` tests start to fail?

I would expect your change in `asdf.sh` to only affect tests in `asdf_sh.bats`, but I don't think there's a test case for this scenarion, so a test should be added there.

Reply:
The `version_commands.bats` tests fail because:
1. They call `setup_asdf_dir` in the setup step.
    1. `setup_asdf_dir` creates a temp directory with no bin or lib files
    2. and sets the `ASDF_DIR` env variable to be the path to that directory.
2. The tests call `. $(dirname "$BATS_TEST_DIRNAME")/asdf.sh`.
3. `asdf.sh` will use the value of the existing `ASDF_DIR` env variable, that was set by  `setup_asdf_dir`
4. and fail because there's no `lib/asdf.sh` file in that `ASDF_DIR`.

This is what I've mentioned in my previous comments, and that's why I've said that "setup_asdf_dir creates an 'empty' ASDF_DIR", and suggested that it should guarantee that the `ASDF_DIR` it creates is complete.